hstream-processing/src/HStream/Processing/Processor.hs

```diff
@@ -207,15 +209,14 @@ runTask statsHolder qid SourceConnectorWithoutCkp {..} sinkConnector taskBuilder
             let acSourceName = iSourceName (taskSourceConfig HM'.! srcStream)
             let (sourceEProcessor, _) = taskTopologyForward HM'.! acSourceName
             liftIO $ updateTimestampInTaskContext ctx srcTimestamp
-            e' <- try $ runEP sourceEProcessor (mkERecord Record {recordKey = srcKey, recordValue = srcValue, recordTimestamp = srcTimestamp})
-            case e' of
-              Left (e :: SomeException) -> liftIO $ do
-                Log.fatal $ Log.buildString (Prelude.show e)
-                query_stat_add_total_execute_errors statsHolder (textToCBytes qid) 1
-              Right _ -> do
-                liftIO $ query_stat_add_total_output_records statsHolder (textToCBytes qid) 1
-                return ()
-          RIO.putMVar mvar ()
+            catches (runEP sourceEProcessor (mkERecord Record {recordKey = srcKey, recordValue = srcValue, recordTimestamp = srcTimestamp}))
+              [ Handler $ \(err :: HE.StreamNotFound) -> do
+                liftIO $ query_stat_add_total_execute_errors statsHolder (textToCBytes qid) 1
+                throw err
+              , Handler $ \(err :: SomeException) -> do
+                liftIO $ Log.warning $ Log.buildString' err
+              ]
```

Comment:
I am unsure if exceptions other than `StreamNotFound` should increase `query_stat_add_total_execute_errors`. Please recheck the definition of the stat item. You can merge the PR if the answer is "no".

Reply:
I prefer to increase `query_stat_add_total_execute_errors` for all exceptions here.